packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/projects/ProjectDuplicateControllerIntegrationTests.java

```diff
@@ -68,9 +110,69 @@ public void beforeEach() {
+        String newProjectId = JsonPath.read(result, "$.data.duplicateProject.project.id");
+
+        TestTransaction.flagForCommit();
+        TestTransaction.end();
+
+        IEMFEditingContext oldProjectEditingContext = getEditingContext(FlowIdentifier.PROJECT_ID);
+        IEMFEditingContext newProjectEditingContext = getEditingContext(newProjectId);
+
+        assertThat(newProjectEditingContext.getDomain().getResourceSet().getResources())
+                .hasSize(oldProjectEditingContext.getDomain().getResourceSet().getResources().size());
+
+        Set<String> oldIds = collectAllIdsOnSiriusResource(oldProjectEditingContext.getDomain().getResourceSet());
+        Set<String> newIds = collectAllIdsOnSiriusResource(newProjectEditingContext.getDomain().getResourceSet());
+
+        assertThat(newIds).as("Both ResourceSet should contain the same number of semantic elements.").hasSize(oldIds.size());
+        assertThat(newIds).as("No semantic id should be shared between the two ResourceSet.").noneMatch(oldIds::contains);
+
+        // Check project name
+        Optional<Project> optProject = projectSearchService.findById(newProjectId);
+        assertThat(optProject).get().extracting(Project::getName).isEqualTo("Flow - Copy");
+
+        //Check that the representation is available.
+        Notifier newSystem = searchByLabel(newProjectEditingContext.getDomain().getResourceSet(), "NewSystem");
+
+        List<RepresentationMetadata> representations = representationMetadataSearchService.findAllRepresentationMetadataBySemanticDataAndTargetObjectId(
+                AggregateReference.to(UUID.fromString(newProjectEditingContext.getId())), identityService.getId(newSystem));
+        assertThat(representations).hasSize(1);
+
+        // Check that all targetObjectIds have been updated
+        Optional<Diagram> diagram = representationSearchService.findById(newProjectEditingContext, representations.get(0).getId().toString(), Diagram.class);
+        assertThat(diagram).isPresent();
+        assertThat(diagram.get().getNodes()).allMatch(node -> objectSearchService.getObject(newProjectEditingContext, node.getTargetObjectId()).isPresent());
+    }
+
+    private Notifier searchByLabel(ResourceSet rs, String expectedLabel) {
+        var contentIte = rs.getAllContents();
+        while (contentIte.hasNext()) {
+            Notifier next = contentIte.next();
+            StyledString label = labelService.getStyledLabel(next);
+            if (label != null && label.toString().equals(expectedLabel)) {
+                return next;
+            }
+        }
+        return null;
+    }
+
+    private IEMFEditingContext getEditingContext(String projectId) {
+        return (IEMFEditingContext) projectEditingContextService.getEditingContextId(projectId)
+                .flatMap(this.editingContextSearchService::findById)
+                .orElseThrow(() -> new InvalidParameterException("Unable to find editing context with id " + projectId));
```

Comment:
What is the link with a `java.security.InvalidParameterException`? There's nothing security related here, just use fail(...)